examples/door-lock-app/linux/src/LockEndpoint.cpp

```diff
@@ -0,0 +1,339 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "LockEndpoint.h"
+#include <cstring>
+
+using chip::to_underlying;
+
+bool LockEndpoint::Lock(chip::Optional<chip::ByteSpan> pin)
+{
+    return setLockState(DlLockState::kLocked, pin);
+}
+
+bool LockEndpoint::Unlock(chip::Optional<chip::ByteSpan> pin)
+{
+    return setLockState(DlLockState::kUnlocked, pin);
+}
+
+bool LockEndpoint::GetUser(uint16_t userIndex, EmberAfPluginDoorLockUserInfo & user) const
+{
+    ChipLogProgress(Zcl, "Door Lock App: LockEndpoint::GetUser [endpoint=%d,userIndex=%hu]", mEndpointId, userIndex);
+
+    uint16_t adjustedUserIndex = static_cast<uint16_t>(userIndex - 1);
```

Comment:
Why are we subtracting 1 here?

Reply:
Per spec, user index changes from 1 to `NumberOfTotalUsersSupported`. I could subtract 1 in the cluster code, but I felt like we should pass the raw values to the app and let the app decide how to store data (maybe they will have a database with index that starts from 1).